src/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComponentActivator.cs

```diff
@@ -0,0 +1,197 @@
+            {
+                throw new ArgumentNullException(nameof(argsNative.delegateType));
+            }
+
+            if (argsNative.functionHandle == IntPtr.Zero)
+            {
+                throw new ArgumentNullException(nameof(argsNative.functionHandle));
+            }
+
+            StringFormat stringFormat = (StringFormat) argsNative.stringFormat;
+
+            CreateDelegateArgs args = new CreateDelegateArgs()
+            {
+                assemblyPath = MarshalToString(stringFormat, argsNative.assemblyPath),
+                typeName     = MarshalToString(stringFormat, argsNative.typeName),
+                methodName   = MarshalToString(stringFormat, argsNative.methodName),
+                delegateType = MarshalToString(stringFormat, argsNative.delegateType),
+            };
+
+            return args;
+        }
+
+        /// <summary>
+        /// Native hosting entry point for creating a native delegate
+        /// </summary>
+        /// <param name="argsNative">Reference to a <see cref="CreateNativeDelegateArgs"/> instance</param>
+        [CLSCompliant(false)]
+        public unsafe static int CreateNativeDelegate(CreateNativeDelegateArgs argsNative)
+        {
+            CreateDelegateArgs args;
+
+            try
+            {
+                args = CheckAndMarshalNativeInput(argsNative);
+
+                Delegate d = CreateDelegate(ref args);
+
+                IntPtr functionPtr = Marshal.GetFunctionPointerForDelegate(d);
+
+                lock(s_Delegates)
+                {
+                    s_Delegates[functionPtr] = d;
+                }
+
+                Marshal.WriteIntPtr(argsNative.functionHandle, functionPtr);
+            }
+            catch (Exception e)
+            {
+                return e.HResult;
+            }
+
+            return 0;
+        }
+
+        /// <summary>
+        /// Native hosting entry point for releasing a created native delegate
+        /// </summary>
+        /// <param name="functionPtr">Native delegate to be released</param>
+        [CLSCompliant(false)]
+        public unsafe static int ReleaseNativeDelegate(IntPtr functionPtr)
+        {
+            try
+            {
+                lock(s_Delegates)
+                {
+                    s_Delegates.Remove(functionPtr);
+                }
+            }
+            catch (Exception e)
+            {
+                return e.HResult;
+            }
+
+            return 0;
+        }
+
+        private static Delegate CreateDelegate(ref CreateDelegateArgs args)
+        {
+            if (PathInternal.IsPartiallyQualified(args.assemblyPath))
+            {
+                throw new ArgumentException(SR.Argument_AbsolutePathRequired, nameof(args.assemblyPath));
+            }
+
+            // Throws
+            Assembly assembly = LoadAssembly(args.assemblyPath);
+
+            using (AssemblyLoadContext.EnterContextualReflection(assembly))
```

Comment:
Would it be better to call `assembly.GetType` below instead?

I look at EnterContextualReflection API as something to use as compat quirk for existing code; not as something that should be used in new code.

Reply:
I would definitely expect `assembly.GetType`.
If there are cases where that is still dependent on the current context, then we should use the overload with the resolver and solve the problem that way.

My understanding is that we propagate `EnterContextualReflection` as a solution for existing code bases which can't be modified. I would epxect us to ask everybody who can to change the code to not rely on context instead.

Given that, we should follow our own guidelines...

Reply:
I was avoiding `assembly.GetType`.  
* It is broken for generics in this exact scenario (framework working on behalf of an ALC) unless using `EnterContextualReflection`.
* It doesn't allow fully assembly qualified type names.
* I didn't want to restrict types to being in the main assembly

I will use the `Type.GetType(resolver:resolver)` overload.